### PR TITLE
Check target boxes input on generalized_rcnn.py

### PR DESCRIPTION
**Pull request by mmcenta**
Solves #2192 with minimal changes, but I can add checks to the `labels` and `scores` fields too. I would also like your opinion on creating a function to make these checks because they can get quite verbose (especially if we check the other fields as well).

Additionally, if we want full validation of the model-specific fields (e.g. `masks` for Mask R-CNN) we need to specify a check function for each child, perhaps by overriding a `_check` method or passing it to the `__init__` method.

### CODE REVIEW

File: torchvision/models/detection/generalized_rcnn.py [hunk 1]
```diff
@@ -57,6 +57,16 @@ def forward(self, images, targets=None):
         """
         if self.training and targets is None:
             raise ValueError("In training mode, targets should be passed")
+        if self.training:
+            boxes = targets["boxes"]
```
Comment (fmassa) [inline torchvision/models/detection/generalized_rcnn.py:61]: `targets` is a list of `dict`, so you might need to iterate over it checking that each element in the list is of the correct shape.
Something in the lines of
```python
assert targets is not None
for target in targets:
    boxes = target["boxes"]
    ...
```
Also, you might need to assert that the `targets` is not None, otherwise torchscript might not be able to infer it.

Comment (mmcenta) [reply torchvision/models/detection/generalized_rcnn.py]: Done!

**Comment by mmcenta** [comment]
I fixed the issue and added tests in `test/models.py`. I chose not to create a new file just yet.
If we choose to expand validation, I can work on the functions and then create separate tests for them.
